### PR TITLE
Draft: Moves all prosemirror deps to peerDependencies & devDependencies

### PR DESCRIPTION
**Pull request by janthurau**
refs  #3209, https://github.com/ueberdosis/hocuspocus/issues/445

### CODE REVIEW

**Comment by janthurau** [comment]
@zanemcca Don't have too much experience with peerDependencies, but I believe this should do the job?

**Review by bdbch — APPROVED** [review]
LGTM, thanks!

File: packages/extension-bubble-menu/package.json [hunk 1]
```diff
@@ -29,12 +29,16 @@
...
+    "prosemirror-view": "^1.28.2"
   },
   "dependencies": {
+    "tippy.js": "^6.3.7",
+    "lodash": "^4.17.21",
+    "@tiptap/core": "^2.0.0-beta.193",
     "prosemirror-state": "^1.4.1",
-    "prosemirror-view": "^1.28.2",
-    "tippy.js": "^6.3.7"
+    "prosemirror-view": "^1.28.2"
```
Comment (zanemcca) [inline packages/extension-bubble-menu/package.json:41]: I anticipate that these three remaining tiptap & prosemirror `dependencies` will still cause trouble. I think if you moved them to `devDependencies` then things should work more smoothly.

**Review by zanemcca — COMMENTED** [review]
Thanks so much for doing this @janthurau! Aside from that one case I noticed I think what you have hear is a fantastic baseline.

I think the next step will be an important one as well.

When you guys upgrade your prosemirror packages in `devDependencies` it would be ideal if the version ranges in the `peerDependencies` were extended to include the new package versions instead of moved forward to match the new `devDependencies`. If you were to move the `peerDepenencies` forward in lock-step with the `devDependencies` then it will force us to use that specific package version even though it may not really be necessary and it could cause trouble with our old editor which was built for an older version.

Obviously if you guys run into breaking changes at specific version of whichever package you can and should remove outdated package versions from the `peerDependencies` but that should be a fairly rare event. I will leave it to you guys to decide how best to mange when to drop support for older packages but the wider the range you can support the easier it is for downstream developers.

**Comment by thet** [comment]
I'm not sure what that change should actually solve.
I'm directly depending on individual tiptap packages. Now I had to add all those prosemirror dependencies to my package dependencies.
Not a huge deal though, but also quite some work to get the build running again.

**Comment by jmamakeesic** [comment]
@bdbch Please see [my comment](https://github.com/ueberdosis/tiptap/commit/30e799c67f2a9d5ef275f965998bcbbeac330226#r97974157) that references a related installation error.

**Comment by bdbch** [comment]
Hey @jmamakeesic - if not already done could you open an issue for that so we can track it better?

**Comment by jmamakeesic** [comment]
https://github.com/ueberdosis/tiptap/issues/3664
https://github.com/ueberdosis/tiptap/pull/3665

**Pull request merged**